File: android/src/main/java/com/lightningkite/khrysalistemplate/layouts/LocationDemoXml.kt
//
// LocationDemoXml.swift
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class LocationDemoXml {

    lateinit var getLocation: Button
    lateinit var locationDisplay: TextView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.location_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        getLocation = view.findViewById<Button>(R.id.get_location)
        locationDisplay = view.findViewById<TextView>(R.id.location_display)
        
        
        
        return view
    }
}

File: web/src/layout/LocationDemoXml.ts
//
// LocationDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './location_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.LocationDemoXml
export class LocationDemoXml {
    xmlRoot!: HTMLElement
    getLocation!: HTMLButtonElement
    locationDisplay!: HTMLElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.getLocation = getViewById<HTMLButtonElement>(view, "get_location")
        this.locationDisplay = getViewById<HTMLElement>(view, "location_display")
        return view
    }
}
